opa-filter-core/src/main/java/com/bisnode/opa/spring/security/filter/OpaFilter.java

```diff
@@ -0,0 +1,72 @@
+package com.bisnode.opa.spring.security.filter;
+
+import com.bisnode.opa.client.OpaClientException;
+import com.bisnode.opa.client.query.OpaQueryApi;
+import com.bisnode.opa.client.query.QueryForDocumentRequest;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.security.access.AccessDeniedException;
+import org.springframework.web.filter.GenericFilterBean;
+
+import javax.servlet.FilterChain;
+import javax.servlet.ServletException;
+import javax.servlet.ServletRequest;
+import javax.servlet.ServletResponse;
+import javax.servlet.http.HttpServletRequest;
+import java.io.IOException;
+
+public class OpaFilter extends GenericFilterBean {
+
+    private static final Logger log = LoggerFactory.getLogger(OpaFilter.class);
+    private final OpaQueryApi opaQueryApi;
+    private final String policyName;
+
+    public OpaFilter(OpaQueryApi opaQueryApi, String policyName) {
+        this.opaQueryApi = opaQueryApi;
+        this.policyName = policyName;
+    }
+
+    @Override
+    public void doFilter(ServletRequest request, ServletResponse response, FilterChain chain) throws IOException, ServletException {
+        if (!(request instanceof HttpServletRequest)) {
```

Comment:
Should we throw exception here, or just continue filtering?

Reply:
Continue filtering, it's a pattern of chain of responsibility.